src/Symfony/Component/Form/Extension/Validator/Constraints/FormValidator.php

```diff
@@ -96,11 +96,19 @@ public function validate($form, Constraint $constraint)
                     ? (string) $form->getViewData()
                     : gettype($form->getViewData());
 
-                $this->context->buildViolation($config->getOption('invalid_message'))
-                    ->setParameters(array_replace(array('{{ value }}' => $clientDataAsString), $config->getOption('invalid_message_parameters')))
+                $failure = $form->getTransformationFailure();
```

Comment:
Sure, but it won't change fundamentally once the logic is extracted from this extension anyway. ^^